olca-app/src/org/openlca/app/editors/graphical/model/GraphFactory.java

```diff
@@ -133,6 +143,7 @@ public static List<Exchange> getExchanges(RootEntity entity, ModelType type) {
 					yield Collections.emptyList();
 				if (p.isFromLibrary()) {
 					Libraries.fillExchangesOf(p);
```

Comment:
@francoislerall `Libraries.fillExchanges` can and should be avoided here, as the quant. reference is already in the database, there is no need for the `libraryProcessExchanges` method then

Reply:
True, I realize that it is also displaying other waste inputs and product outputs than the quantitative reference. And this is something we do not want, right?

Reply:
then these are multi-functional processes and libraries should not contain such processes, these flows should not be hidden then in the graph because they indicate a library error

Reply:
I am not sure then how to display all the multifunctional provider flows without using the `Libraries.fillExchanges`?

What I have in mind is the following:
```java
if (p.isFromLibrary()) {
	Libraries.fillExchangesOf(p);
	var products = AllocationUtils.getProviderFlows(p);
	if (products.size() > 1) {
		log.warn("Displaying a multi-functional process that should not" +
				" belong to the " + p.library);
		yield libraryProcessExchanges(p);
	}
	yield p.quantitativeReference != null
			? Collections.singletonList(p.quantitativeReference)
			: Collections.emptyList();
}
```

Reply:
Not sure if I understand, but I think library processes can be just treated like normal processes, they already contain their reference flow (and other product outputs or waste inputs in case of multi-functional process mistakes). So, you do not need to fill or filter these processes.

Reply:
Maybe, you did not want to say this:
> these flows should not be hidden then in the graph
?

Reply:
it was an answer to this:

> True, I realize that it is also displaying other waste inputs and product outputs than the quantitative reference. And this is something we do not want, right?

not need to filter these, they should be displayed